### ISSUE TITLE
⚠️ Testnet is down (unknown catchain-related disastrous bug)

### ISSUE THREAD

**Issue by Skydev0h** [ton-blockchain/ton#272]
Testnet has crashed approximately at 19:02:13 +0200.
I do not know exact reason but after starting up next **FATAL** occurs:
```
[catchain-receiver.cpp:585][!catchainreceiver] Check `last_sent_block_->delivered()` failed
```

If I purge catchain receiver data, next error happens:
```
[catchain-receiver.cpp:316][!catchainreceiver]   Check `!S->blamed()` failed
```

Also, there are lots of warnings in logs about incorrect signature of left fork blame, "CAIN: blaming source", a CATCHAIN_WARNING that it "got ill" (validator caught coronavirus?!).

Also there was information that one of validators got full disk nearly when this happened.

The network seems to be still down now (at least my validator tries to download something, state or catchain, and then fatals out).

**Comment by Skydev0h** [comment]
![image](https://user-images.githubusercontent.com/5331782/77370955-bfcc0000-6d6a-11ea-9cff-23627c18797e.png)
It is scary, it seems that the pandemic have reached the TON testnet

**Comment by Skydev0h** [comment]
![image](https://user-images.githubusercontent.com/5331782/77371241-8942b500-6d6b-11ea-84ba-382ddffe5cd8.png)
The moment before **fatal**

**Comment by Skydev0h** [comment]
After last emerg fix validation suddenly stops after some seconds of seemingly successful validation:
![image](https://user-images.githubusercontent.com/5331782/77373734-a67a8200-6d71-11ea-9ab7-9babccfd0867.png)
![image](https://user-images.githubusercontent.com/5331782/77373747-ac706300-6d71-11ea-8f89-9d213b637d45.png)
And last block is 22k seconds ago.

**Comment by Skydev0h** [comment]
![image](https://user-images.githubusercontent.com/5331782/77409225-8b376300-6dc1-11ea-9f05-15c26d591299.png)
It seems that validation is working as of now, but the state of config and elector is very strange.

**Comment by ton-blockchain** [comment]
Well, recovering from the situation when somebody controlling 40% of testnet validators upgraded them to a buggy version that corrupted the local database, including current catchain state, is not simple. However, we see that this turned out to be possible, even though it required an emergency update of more than 2/3 of all validators. So let's think of this as an unplanned test of the behavior of the TON Blockchain in such harsh conditions.

Such things are unlikely to happen in the mainnet, because nobody is expected to control such a large portion of the validators there, so they would not have been upgraded to a buggy version simultaneously.

On the flip side, we see that no forks and no invalid blocks have been created during the several hours when the testnet was not working properly, so the Catchain consensus turned out to be robust enough to cope with this situation (as it should), even if it generated a lot of error messages in the process. We still value C and A of the CAP-theorem higher than P.

**Comment by Skydev0h** [comment]
Wow, that was a bit unexpected. I tracked owners by wallet addresses and did not see anyone near 40% on any voting.
![image](https://user-images.githubusercontent.com/5331782/77468916-f6f7eb00-6e16-11ea-80ed-9fe2a08db95f.png)
This was the last log before last voting ended before the crash.

It seems that your validators have 624 278 weight out of 1 277 679 (48.86%), if subtract 18.78% (my validators used latest git commit and were not patched in any way) then 32.36% remains.

I rechecked those percentages in "current validators" section after switch:
![image](https://user-images.githubusercontent.com/5331782/77469897-79cd7580-6e18-11ea-84e6-97170fb4b683.png)

That means that someone controls all other validators?
